### PR TITLE
ETH -> Elrond part 3

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added new components and changed existing ones as to allow the eth->elrond flow to work as expected
- fixed a possible panic in ElrondClient

### CODE REVIEW

File: integrationTests/mock/contracts/elrond.go [hunk 1]
```diff
@@ -18,33 +20,40 @@ type ElrondContract struct {
...
 
 func (ec *ElrondContract) createContractFunctions() {
 	ec.AddHandler("getCurrentTxBatch", ec.getCurrentTxBatch)
 	ec.AddHandler("userRole", ec.userRole)
+	ec.AddHandler("getTokenIdForErc20Address", ec.getTokenIdForErc20Address)
 }
 
 func (ec *ElrondContract) getCurrentTxBatch(caller string, value string, arguments ...string) ([][]byte, error) {
-	log.Warn("getCurrentTxBatch", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	log.Debug("getCurrentTxBatch", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
```
Comment (bogdan-rosianu) [inline integrationTests/mock/contracts/elrond.go:50]: I think the logger will automatically do a `fmt.Sprintf("%v")` over `arguments`

File: integrationTests/mock/contracts/elrond.go [hunk 2]
```diff
@@ -58,6 +67,26 @@ func (ec *ElrondContract) userRole(caller string, value string, arguments ...str
 	return [][]byte{userRole}, nil
 }
 
+func (ec *ElrondContract) getTokenIdForErc20Address(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Debug("getTokenIdForErc20Address", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+
+	if len(arguments) != 1 {
+		return nil, ErrInsufficientArguments
```
Comment (bogdan-rosianu) [inline integrationTests/mock/contracts/elrond.go:74]: maybe there are too many arguments. in this case, the error message is misleading

File: integrationTests/mock/contracts/errors.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package contracts
+
+import "errors"
+
+// ErrNilTokenHandler signals that a nil token handler instance has been provided
+var ErrNilTokenHandler = errors.New("nil TokenHandler instance")
+
+// ErrInsufficientArguments signals that insufficient arguments were provided
+var ErrInsufficientArguments = errors.New("insufficient argument")
```
Comment (bogdan-rosianu) [inline integrationTests/mock/contracts/errors.go:9]: `insufficient arguments` . also, as stated in the previous comment, I would name it `ErrInvalidNumberOfArguments` or something similar

File: integrationTests/mock/contracts/elrond.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/mock/contracts/elrond.go:50]: done

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**